### PR TITLE
BaSP-TG-10: Create and get Tasks.

### PR DESCRIPTION
**Pull request by EliasBermudez13**
Add a new task to the MongoDB database using Joi validations and Mongoose Schema.
Get all the tasks from the database. Get only one finded by its "id".

### CODE REVIEW

File: src/controllers/tasks.js [hunk 1]
```diff
@@ -1,91 +1,80 @@
...
-        foundTask.description = updateTask.description ?? task.description;
-        foundTask.project_id = updateTask.project_id ?? task.project_id;
-        foundTask.hours = updateTask.hours ?? task.hours;
+const createTask = async (req, res) => {
+  try {
+    const task = new Tasks({
+      name: req.body.name,
+      description: req.body.description,
+      project_id: req.body.project_id,
+      hours: req.body.hours,
```
Comment (martinraveglia) [inline src/controllers/tasks.js:59]: this is not as the data model that was provided to the team. Update it taking into account that data model

File: src/models/Tasks.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const taskSchema = new Schema({
+  name: { type: String, required: true },
+  description: { type: String, required: true },
+  project_id: mongoose.SchemaTypes.ObjectId,
+  hours: { type: Number, required: true },
```
Comment (martinraveglia) [inline src/models/Tasks.js:9]: this is not as the data model that was provided to the team. Update it taking into account that data model

File: src/validations/tasks.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import Joi from 'joi';
+
+const validateCreation = (req, res, next) => {
+  const tasksValidation = Joi.object({
+    name: Joi.string().min(3).max(50).required(),
+    description: Joi.string().min(3).max(50).required(),
+    project_id: Joi.required(),
+    hours: Joi.number().required(),
```
Comment (martinraveglia) [inline src/validations/tasks.js:8]: this is not as the data model that was provided to the team. Update it taking into account that data model

File: src/validations/tasks.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      data: undefined,
+      error: true,
+    });
+  }
+  return next();
+};
+
+export default {
+  validateCreation,
+};
```
Comment (martinraveglia) [inline src/validations/tasks.js:25]: if we are exporting this with default, there is no need to export it as an object. Export the plain method instead
```suggestion
export default validateCreation;
```

**Review by martinraveglia — CHANGES_REQUESTED** [review]

File: src/controllers/tasks.js (hunk 1 above)
Comment (EliasBermudez13) [reply src/controllers/tasks.js:59]: I'm sorry i don't get it, can you explain yourself a little more? Thanks

File: src/index.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-import routes from './routes';
+import taskRoutes from './routes/tasks';
 
 const app = express();
 const port = process.env.PORT || 3000;
 const MONGO_URL = 'mongodb+srv://RadiumA:RadiumA@trackgenix.r6u6do6.mongodb.net/?retryWrites=true&w=majority';
 
 app.use(express.json());
-app.use(routes);
+app.use('/tasks', taskRoutes);
```
Comment (martinraveglia) [inline src/index.js:10]: This belongs to the routes index file!!

**Review by martinraveglia — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 5 above)
Comment (EliasBermudez13) [reply src/index.js:10]: Like that ? : app.use (taskRoutes);

Comment (EliasBermudez13) [reply src/index.js:10]: Or i left only app.use(routes)

Comment (martinraveglia) [reply src/index.js:10]: you should let app.use(routes) here in this file, take a look at the routes index.js to understand why

Comment (EliasBermudez13) [reply src/index.js:10]: okay, thanks

File: src/index.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import express from 'express';
 import mongoose from 'mongoose';
-import routes from './routes';
+import routes from './routes/index';
```
Comment (martinraveglia) [inline src/index.js:3]: this file should be unmodified, originally this line was imported as,
```suggestion
import routes from './routes';
```

**Review by martinraveglia — CHANGES_REQUESTED** [review]

File: postman_collection.json [hunk 7]
```diff
@@ -24,7 +24,73 @@
...
+					"response": []
+				},
+				{
+					"name": "Post",
+					"request": {
+						"method": "POST",
+						"header": [],
+						"body": {
+							"mode": "raw",
+							"raw": "{\r\n    \"name\": \"Task3\",\r\n    \"description\": \"One more task descritpion for test purposes\",\r\n    \"project_id\": \"634de88a9a45e5e780e62787\",\r\n    \"hours\": 1\r\n}   ",
```
Comment (martinraveglia) [inline postman_collection.json:54]: this body data is not updated, more data than necessary is being sent

**Review by martinraveglia — CHANGES_REQUESTED** [review]

**Review by martinraveglia — APPROVED** [review]

File: src/routes/tasks.js [hunk 8]
```diff
@@ -1,9 +1,15 @@
 import express from 'express';
-import { deleteTask, editTask } from '../controllers/tasks';
-import validateEdit from '../validations/tasks';
+import { getAllTasks, getTaskById, createTask, deleteTask, editTask } from '../controllers/tasks';
+import {validateCreation, validateEdit} from '../validations/tasks';
```
Comment (martinraveglia) [inline src/routes/tasks.js:3]:
```suggestion
import { validateCreation, validateEdit } from '../validations/tasks';
```

File: src/validations/tasks.js [hunk 9]
```diff
@@ -17,4 +33,6 @@ const validateEdit = (req, res, next) => {
   return next();
 };
 
-export default validateEdit;
+export{ validateCreation ,
+ validateEdit 
+};
```
Comment (martinraveglia) [inline src/validations/tasks.js:38]:
```suggestion
export { 
 validateCreation,
 validateEdit 
};
```

**Review by martinraveglia — CHANGES_REQUESTED** [review]

**Review by martinraveglia — APPROVED** [review]

**Pull request merged**